### PR TITLE
update: First set of ATB updates

### PR DESCRIPTION
**Pull request by mrshu**
* Add ATB to generated URL

* Add CSS files

* Add the confetti JS file

Signed-off-by: mr.Shu <mr@shu.io>

---------------------------------------------------------

@bsstoner a couple of notes on what works and what does not:

- the ATB param works. The only issue is that we have no idea of knowing whether the extension is currently newly installed or not. I am therefore checking for the existence of `localStorage['last_version']` -- something I have used to determine whether the extension has updated -- to determine whether it is a new installation or not. Note that this might be inaccurate as the particular section of code which used this key of `localStorage` has been removed 6 months ago.

In order to try it out, though, you need to clear `localStorage`. I remember that a long long time ago you have shown me how to do that but I just could not do it now. If you could recall the exact procedure, I would very much appreciate the details.

- The injection of JS/CSS in Safari is rather simplistic. You inject everything into every domain you list (and you have to list everything, see the PR) -- there is no other way. This is quite nasty. While I do not think there will be an element with a class  `ddg-extension-hide` or one with an id `atb-extension-overlay` on Google, there might (at some point) be a class `.gb_g` on DuckDuckGo. 

I am not sure how relevant this is but I thought you should know. 

- I did not try to do anything with the confetti, but at least prepared the file for you. In this case, no messages will be sent. While I could not find it in the [docs](https://developer.apple.com/library/safari/documentation/Tools/Conceptual/SafariExtensionGuide/InjectingScripts/InjectingScripts.html) I've tested this myself and it seems that these scripts are also injected to pages which are already loaded (so the system you've put together should work).

That is all I have, please do ask about anything I forgot.

### CODE REVIEW

File: duckduckgo.safariextension/Info.plist [hunk 1]
```diff
@@ -66,11 +66,398 @@
...
+			<string>https://*.google.co.ve/*</string>
+			<string>https://*.google.vg/*</string>
+			<string>https://*.google.co.vi/*</string>
+			<string>https://*.google.com.vn/*</string>
+			<string>https://*.google.vu/*</string>
+			<string>https://*.google.ws/*</string>
+			<string>https://*.google.rs/*</string>
+			<string>https://*.google.co.za/*</string>
+			<string>https://*.google.co.zm/*</string>
+			<string>https://*.google.co.zw/*</string>
```
Comment (bsstoner) [inline duckduckgo.safariextension/Info.plist]: wow, that's annoying.

File: duckduckgo.safariextension/html/global.html [hunk 2]
```diff
@@ -117,6 +117,15 @@
...
     safari.application.addEventListener("contextmenu", ddgContextMenu, false);
 
+    // use previously used localStorage['last_version'] to determine whether
+    // this is a new install or not
+    // https://github.com/duckduckgo/safari-zeroclickinfo/commit/4746f30668033551025847a33234ad15579f53d5#diff-89d925e2bf34fb6e3d713637cfd75bdbL39
+    if (typeof localStorage['last_version'] == "undefined") {
+      if (typeof localStorage['atb'] == "undefined") {
+        localStorage['atb'] = 'v' + Math.ceil((new Date().getTime() - 1456333200000) / 604800000);
+      }
+    }
```
Comment (bsstoner) [inline duckduckgo.safariextension/html/global.html]: I can't figure out where the localStorage data is. This page says it's stored on the domain where the extension is injected into: https://developer.apple.com/library/safari/documentation/Tools/Conceptual/SafariExtensionGuide/ExtensionSettings/ExtensionSettings.html

That doesn't appear to be the case though. I see it adding the atb param though so I know it's working...

Comment (mrshu) [inline duckduckgo.safariextension/html/global.html]: @bsstoner let me see if I can help.

The `localStorage['atb']` is actually always contained in this `global.html` page -- it never goes out. The addition of the ATB param also happens in here, on line 148 and below. The page you linked also says

> When used from a global HTML page or extension bar, the domain of the local storage is the extension. This data belongs to your extension.

I believe this is the case here.

Did that help?

Comment (bsstoner) [inline duckduckgo.safariextension/html/global.html]: Yea, I'm trying to read through their docs now to understand this madness. Where is global.html loaded? Can you see it from the Web Inspector?

File: duckduckgo.safariextension/Info.plist [hunk 3]
```diff
@@ -66,11 +66,398 @@
...
 		</dict>
+		<key>Stylesheets</key>
+		<array>
+			<string>css/noatb.css</string>
+			<string>css/nopopup.css</string>
+		</array>
+		<key>Whitelist</key>
+		<array>
+			<string>https://*.duckduckgo.com/*</string>
+			<string>https://duckduckgo.com/*</string>
```
Comment (bsstoner) [inline duckduckgo.safariextension/Info.plist]: Maybe worth adding http entries for DDG too? In case people change the setting?

**Comment by bsstoner** [comment]
@mrshu I think this is ready to merge.

Are you aware of anything else that needs to be updated to account for that fact that I used my certificate instead of yours?

**Comment by mrshu** [comment]
@bsstoner the only thing I am aware of is the `DeveloperIdentifier` in both `duckduckgo.plist` and  `Info.plist` (inside duckduckgo.safariextz) -- assuming that that has changed (e.g. you see a different value in your Extension Builder).

Othe than that this looks good to me!

**Comment by bsstoner** [comment]
I think it's actually the same:
screen shot 2016-08-18 at 9 01 06 pm

**Pull request merged**